pkg/cluster/gossip.go

```diff
@@ -123,8 +123,17 @@ func (c *GossipConfig) ApplyDefaults() error {
 			return fmt.Errorf("bootstrapping peer discovery: %w", err)
 		}
 		c.discoverer = d
+	}
+
+	return c.RefreshPeers()
+}
 
-		addrs, err := d.Addrs(c.DiscoverPeers, stdlog.New(io.Discard, "", 0)) // TODO(rfratto): log to log.Logger?
+// RefreshPeers updates the list of peers the node will look to connect to.
+func (c *GossipConfig) RefreshPeers() error {
```

Comment:
This requires ApplyDefaults to be called before calling RefreshPeers, otherwise `c.discoverer` may be nil. 

Can you find a way to either:

1. Change the semantics to avoid c.discoverer being nil, or 
2. Document that ApplyDefaults _must_ be called?

(IMO it's a bit weird that the config is responsible for discovering and refreshing peers, so it might be possible to change the semantics to introduce some separation of concerns)

Reply:
I've documented that ApplyDefaults must be called successfully before using it to create a NewGossipNode.